route-rs-runtime/src/link/classify_link.rs

```diff
@@ -1,64 +1,140 @@
 use crate::element::Classifier;
 use crate::link::task_park::*;
-use crate::link::{QueueEgressor, PacketStream};
+use crate::link::{Link, LinkBuilder, PacketStream, QueueEgressor};
 use crossbeam::atomic::AtomicCell;
 use crossbeam::crossbeam_channel;
 use crossbeam::crossbeam_channel::{Receiver, Sender};
 use futures::{Async, Future, Poll, Stream};
 use std::sync::Arc;
 
-pub struct ClassifyLink<'a, C: Classifier> {
-    pub ingressor: ClassifyIngressor<'a, C>,
-    pub egressors: Vec<QueueEgressor<C::Packet>>,
+pub struct ClassifyLink<C: Classifier> {
+    in_stream: Option<PacketStream<C::Packet>>,
+    classifier: Option<C>,
+    dispatcher: Option<Box<dyn Fn(C::Class) -> usize + Send + Sync + 'static>>,
+    queue_capacity: usize,
+    branches: Option<usize>,
 }
 
-impl<'a, C: Classifier> ClassifyLink<'a, C> {
-    pub fn new(
-        input_stream: PacketStream<C::Packet>,
-        classifier: C,
-        dispatcher: Box<dyn Fn(C::Class) -> usize + Send + Sync + 'a>,
-        queue_capacity: usize,
-        branches: usize,
+impl<C: Classifier> ClassifyLink<C> {
```

Comment:
We can also define an `ingressor` function on `ClassifyLink` that accepts only 1 `PacketStream`.

Reply:
done